packages/shared/src/hooks/squads/useSquadActions.ts

```diff
@@ -0,0 +1,17 @@
+import { useMemo } from 'react';
+import { useMutation, useQueryClient } from 'react-query';
+import { updateSquadMemberRole } from '../../graphql/squads';
+
+interface UseSquadActions {
+  onUpdateRole?: typeof updateSquadMemberRole;
+}
+
+export const useSquadActions = (): UseSquadActions => {
+  const client = useQueryClient();
+  const { mutateAsync: onUpdateRole } = useMutation(updateSquadMemberRole, {
+    onSuccess: (_, { sourceId }) =>
+      client.invalidateQueries(['squadMembers', sourceId]),
+  });
+
+  return useMemo(() => ({ onUpdateRole }), []);
```

Comment:
`onUpdateRole` missing in dependency array could lead to stale function references.

Reply:
Right. But that function doesn't really depend on anything since it just accepts a parameter and executes functions. 

Having it on the dependency list and not being a memoized callback would just re-render it every time and make the use memo not fruitful. Or I could be wrong. WDYT?

Reply:
Yeah but like this you get the false sense of stable function which could then be stale due to inner workings of a hook. In that case it is better to skip useMemo all together. 

Also I do think that mutation reference might be stable in `useMutation`, but you would need to check that. If it is not then I would just avoid useMemo in this case.

Reply:
>  In that case it is better to skip useMemo all together.

Totally agree with this. It has just become a thing that our hooks are memoized. Probably we can talk more about this in our next web team meeting to reconsider when we should use it.